hikey.mk

```diff
@@ -32,38 +47,34 @@ EDK2_BIN 			?= $(EDK2_PATH)/Build/HiKey/RELEASE_GCC49/FV/BL33_AP_UEFI.fd
 EDK2_BUILD			?= RELEASE
 endif
 
-LINUX_CONFIG_ADDLIST		?= $(LINUX_PATH)/kernel.config
-
-MCUIMAGE_BIN			?=$(ROOT)/out/mcuimage.bin
-USBNETSH_PATH			?=$(ROOT)/out/usbnet.sh
+MCUIMAGE_BIN			?=$(EDK2_PATH)/HisiPkg/HiKeyPkg/NonFree/mcuimage.bin
 STRACE_PATH			?=$(ROOT)/strace
 BOOT_IMG			?=$(ROOT)/out/boot-fat.uefi.img
 LLOADER_PATH			?=$(ROOT)/l-loader
 NVME_IMG			?=$(ROOT)/out/nvme.img
+OUT_PATH			?=$(ROOT)/out
+GRUB_PATH			?=$(ROOT)/grub
+PATCHES_PATH			?=$(ROOT)/patches_hikey
```

Comment:
For this, I considered 3 choices:

1. Originally the required patch/helper files were included in this makefile itself, i.e. created using echo commands. As more files are required, the makefile becomes really long and looks cluttered, so it does not seem to scale.

2. Add the patch/helper files to this repo under a ```hikey``` dir as you'd mentioned, but there's already a ```kconfigs``` dir here that contains ```hikey.conf``` among other things, so this seems like it might clutter up the repo as well and we'll have to go to multiple locations/paths for the files.

3. Yea another Git tree. This was chosen as it was already done for the mtk platform, and I'd already gone down this path before ```kconfigs``` was created :s.

When the ```kconfigs``` PR (32) was proposed, I thought I could maybe utilize it using some structure like below:
```
build/patches
    |-- hikey
        |-- dir1
        |-- dir1
    |-- mtk
        |-- dir1
        |-- dir2
```
OR
```
build
    |-- hikey
        |-- patches
            |-- dir1
            |-- dir2
    |-- mtk
        |-- patches
            |-- dir1
            |-- dir2
```
This would be similar to 2 above and more generic, but the PR only added a ```kconfigs``` dir, so I couldn't really fit all the patch/helper files there.

Perhaps 3 can be something to consider in the future?